tests/e2e/replication_experimental/multitenancy.py

```diff
@@ -0,0 +1,1508 @@
+    except mgclient.DatabaseError:
+        failed = True
+    assert failed
+
+
+def test_multitenancy_drop_and_recreate_while_replica_using(connection):
+    # Goal: to show that replication can be established against REPLICA where a new databases
+    # needs replicating over
+    # 0/ Setup replication
+    # 1/ MAIN CREATE DATABASE A
+    # 2/ Write to MAIN A
+    # 3/ Validate replication of changes to A have arrived at REPLICA
+    # 4/ Start A transaction on replica 1, Use A on replica2
+    # 5/ Check that the drop/create replicated
+    # 6/ Validate that the transaction is still active and working and that the replica2 is not pointing to anything
+
+    # 0/
+    interactive_mg_runner.start_all(MEMGRAPH_INSTANCES_DESCRIPTION)
+    main_cursor = connection(7687, "main").cursor()
+
+    # 1/
+    execute_and_fetch_all(main_cursor, "CREATE DATABASE A;")
+
+    # 2/
+    execute_and_fetch_all(main_cursor, "USE DATABASE A;")
+    execute_and_fetch_all(main_cursor, "CREATE (:Node{on:'A'});")
+
+    # 3/
+    def retrieve_data():
+        execute_and_fetch_all(main_cursor, "USE DATABASE A;")
+        return set(execute_and_fetch_all(main_cursor, "SHOW REPLICAS;"))
+
+    expected_data = {
+        ("replica_1", "127.0.0.1:10001", "sync", 1, 0, "ready"),
+        ("replica_2", "127.0.0.1:10002", "async", 1, 0, "ready"),
+    }
+    actual_data = mg_sleep_and_assert(expected_data, retrieve_data)
+    assert actual_data == expected_data
+
+    # 4/
+    replica1_cursor = connection(7688, "replica").cursor()
+    replica2_cursor = connection(7689, "replica").cursor()
+
+    execute_and_fetch_all(replica1_cursor, "USE DATABASE A;")
+    execute_and_fetch_all(replica1_cursor, "BEGIN")
+    execute_and_fetch_all(replica2_cursor, "USE DATABASE A;")
+
+    execute_and_fetch_all(main_cursor, "USE DATABASE memgraph;")
+    execute_and_fetch_all(main_cursor, "DROP DATABASE A;")
+
+    # 5/
+    execute_and_fetch_all(main_cursor, "CREATE DATABASE A;")
+    execute_and_fetch_all(main_cursor, "USE DATABASE A;")
+
+    def retrieve_data():
+        return set(execute_and_fetch_all(main_cursor, "SHOW REPLICAS;"))
+
+    expected_data = {
+        ("replica_1", "127.0.0.1:10001", "sync", 0, 0, "ready"),
+        ("replica_2", "127.0.0.1:10002", "async", 0, 0, "ready"),
+    }
+    actual_data = mg_sleep_and_assert(expected_data, retrieve_data)
+    assert actual_data == expected_data
+
+    # 6/
+    assert execute_and_fetch_all(replica1_cursor, "MATCH(n) RETURN count(*);")[0][0] == 1  # one node
+    execute_and_fetch_all(replica1_cursor, "COMMIT")
+    failed = False
+    try:
+        execute_and_fetch_all(replica1_cursor, "MATCH(n) RETURN n;")
+    except mgclient.DatabaseError:
+        failed = True
+    assert failed
+
+    failed = False
+    try:
+        execute_and_fetch_all(replica2_cursor, "MATCH(n) RETURN n;")
+    except mgclient.DatabaseError:
+        failed = True
+    assert failed
```

Comment:
Why this fails if database is still there, it was dropped and created, right?

Reply:
Important to note: They are semantically different databases.

If you have a session on REPLICA for database X, but database X is dropped from RPC given by MAIN. Then the session, once command has committed/rolledback, is now no longer associated with that database X. 
Hence any query requiring a database should DatabaseError.
If MAIN creates X it is, and should, be considered a different database. It would be problematic for that session to allow the user to continue on the new database as if it was the old one.